src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -368,23 +366,59 @@ def _check_private_endpoint(cmd, registry_name, vnet_of_private_endpoint):  # py
                                                subscription=res['subscription'])
 
     # retrieve FQDNs for registry and its data endpoint
+    from .aaz.latest.network.nic import Show as _NICShow
+    from .aaz.latest.network.private_endpoint import Show as _PEShow
+
     pe_ids = [e.private_endpoint.id for e in registry.private_endpoint_connections if e.private_endpoint]
     dns_mappings = {}
+
+    class PEShow(_PEShow):
+        @classmethod
+        def _build_arguments_schema(cls, *args, **kwargs):
+            from azure.cli.core.aaz import AAZStrArg
+            args_schema = super()._build_arguments_schema(*args, **kwargs)
+            args_schema.subscription = AAZStrArg()
+            return args_schema
+
+        def pre_operations(self):
+            args = self.ctx.args
+            # cross subscription
+            self.ctx.update_aux_subscriptions(args.subscription)
```

Comment:
After rethink, I found this is the only correct way.
```suggestion
            self.ctx._subscription_id = args.subscription
```